From cc603d81141bfd2d87de7c3f09ae63bed20f4d7e Mon Sep 17 00:00:00 2001
From: Ivan Pavlotsky <ivan.pavlotsky@lge.com>
Date: Tue, 13 Oct 2026 23:53:24 +0300
Subject: [PATCH] [build][webos] 156.0.8078.0~1: V8: Add webOS targets to
 :v8_snapshot visibility list

In the upstream V8 CL
https://chromium-review.googlesource.com/c/v8/v8/+/8474951, which was
rolled into Chromium in the upstream CL http://crrev.com/c/8473144,
moved :v8_snapshot from public_deps to deps. It caused configuration
errors on building webruntime for webOS/OSE:
| ERROR Input to targets not generated by a dependency.
| The file:
|   ../work/raspberrypi4_64-webos-linux/webruntime-clang/155.0.8059.0/webruntime-clang-155.0.8059.0/Release/snapshot_blob.bin
| is listed as an input or source for the targets:
|   //webos/install/wam_demo:weboswebruntime
|   //webos/install/browser_shell:weboswebruntime
|   //webos/install/weboswebruntime:weboswebruntime
|   //webos/install/default_browser:default_browser
| but this file was not generated by any dependencies of the targets. The target
| that generates the file is:
|   //v8:run_mksnapshot_default

In order to fix these errors:
1. Replaced general :v8 dep with the concrete :v8_snapshot dep in
   //webos/install/* targets.
2. Added //webos/install/* targets to :v8_snapshot visibility list

Issue: NEVA-11481
---
 BUILD.gn | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index cc9788a7fd5..18f80a33f97 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -3511,6 +3511,15 @@ v8_source_set("v8_snapshot") {
   # Let external targets depend on v8_snapshot.
   if (v8_use_external_startup_data) {
     visibility = [ ":*" ]  # Targets in this file can depend on this.
+    # NOTE(neva): For snapshot_blob.bin
+    if (is_webos) {
+      visibility += [
+        "//webos/install/weboswebruntime:weboswebruntime",
+        "//webos/install/wam_demo:weboswebruntime",
+        "//webos/install/browser_shell:weboswebruntime",
+        "//webos/install/default_browser:default_browser",
+      ]
+    }
   }
 
   deps = [
-- 
2.34.1

